### DISCUSSION TITLE
Support macOS (Darwin)

### DISCUSSION THREAD

**Discussion by istvanzk** [Jakeler/ble-serial#34] [Ideas]
Hi,

Thank you for the nice tool.
I wanted to run this on macOS (Darwin) but this was not supported 'out-of-the-box'.

Luckily, the required code change to enable this is very simple.
In ble_serial/__init__.py, I have modified line 3 to
`if (platform.system() == 'Linux') or (platform.system() == 'Darwin'):`

Also, the ReadMe could be updated to indicate that when running on macOS, the BT MAC address is not displayed by blue-scan and instead an OS specific UUID is displayed, which then can be used with the option -d.

I have succesfully tested the above in macOS 10.15.7 (Catalina).

Best

**Comment by Jakeler** [comment]
Thanks for trying it out on macOS! That sounds really promising.
Can you post a example how it the scan output looks with this specific MAC/ID? 
I assume this only applies to the device address and the characteristic UUIDs from the deep scan are still the same as on other platforms?

**Reply by istvanzk** [reply]
> ble-scan
```
Started BLE scan

55AEB150-5AF1-4318-A02E-19A46223F572 (RSSI=-70): nRF52 Accelerometer
43D44058-B486-4013-9836-764FBCAA9F18 (RSSI=-89): Unknown
37D96B1D-6340-4F61-9CD9-3BCBAF72A51F (RSSI=-92): Unknown

Finished BLE scan
```
> ble-scan -d 55AEB150-5AF1-4318-A02E-19A46223F572
```
Started deep scan of 55AEB150-5AF1-4318-A02E-19A46223F572

SERVICE 00000001-af0e-4c28-95a4-4509fd91e0bb (Handle: 10): SDP
     CHARACTERISTIC 00000006-af0e-4c28-95a4-4509fd91e0bb (Handle: 11): Unknown ['read', 'notify']
         DESCRIPTOR 00002902-0000-1000-8000-00805f9b34fb (Handle: 13): Client Characteristic Configuration
     CHARACTERISTIC 00000005-af0e-4c28-95a4-4509fd91e0bb (Handle: 14): TCS-BIN ['write-without-response', 'notify']
         DESCRIPTOR 00002902-0000-1000-8000-00805f9b34fb (Handle: 16): Client Characteristic Configuration

Completed deep scan of 55AEB150-5AF1-4318-A02E-19A46223F572
```
The `55AEB150-5AF1-4318-A02E-19A46223F572` is the OS specific UUID displayed instead of the MAC address. This is also the behaviour as described in the documentation for [Bleak Scan/Discover](https://bleak.readthedocs.io/en/latest/scanning.html) operation.

The BLE characteristic UUIDs `00000006-af0e-4c28-95a4-4509fd91e0bb` and `00000005-af0e-4c28-95a4-4509fd91e0bb` are the ones to be used in the blue-serial:
> ble-serial -d 55AEB150-5AF1-4318-A02E-19A46223F572 -w 00000005-af0e-4c28-95a4-4509fd91e0bb -r 00000006-af0e-4c28-95a4-4509fd91e0bb

```
20:57:01.833 | INFO | linux_pty.py: Slave created on /tmp/ttyBLE -> /dev/ttys000
20:57:01.833 | INFO | ble_interface.py: Receiver set up
20:57:01.833 | INFO | ble_interface.py: Trying to connect with 55AEB150-5AF1-4318-A02E-19A46223F572
20:57:06.604 | INFO | ble_interface.py: Device 55AEB150-5AF1-4318-A02E-19A46223F572 connected
20:57:06.605 | INFO | ble_interface.py: Found write characteristic 00000005-af0e-4c28-95a4-4509fd91e0bb (H. 14)
20:57:06.605 | INFO | ble_interface.py: Found notify characteristic 00000006-af0e-4c28-95a4-4509fd91e0bb (H. 11)
20:57:07.099 | INFO | main.py: Running main loop!
```

**Reply by Jakeler** [reply]
Thanks again! I have changed the platform check now and added macOS examples/description to the readme. 
Released in [v2.3.0](https://github.com/Jakeler/ble-serial/releases/tag/v2.3.0).
